server/src/controllers/package.controller.ts

```diff
@@ -19,8 +21,18 @@ export const packageController = {
 
         res.status(200).send(generatedPackage);
     },
-    streamPackageGeneration: async (req: Request<any, any, PackagesGenerationParams>, res: Response) => {
-        const { data: validatedBody, error } = PackagesGenerationParamsSchema.safeParse(req.body);
+    streamPackageGeneration: async (req: Request<any, any, PackagesGenerationParamsWithFreeText>, res: Response) => {
+        const { freeText, ...restInitial } = req.body;
+        let rest: any = restInitial;
+
+        if (freeText) {
+            packagesLogger.info(`💬 Received free text input: ${freeText}`);
+            rest = await packageService.transformFreeTextIntoPackagesGenerationParams(freeText);
+            packagesLogger.info(`✨ Transform the free text input into: ${JSON.stringify(rest)}`);
+        }
```

Comment:
```suggestion
    streamPackageGeneration: async (
        req: Request<any, any, PackagesGenerationParamsWithStringDatesAndFreeText>,
        res: Response
    ) => {
        const { freeText, ...initialParams } = req.body;
        let params = initialParams;

        if (freeText) {
            packagesLogger.info(`💬 Received free text input: ${freeText}`);
            params = await packageService.transformFreeTextIntoPackagesGenerationParams(freeText);
            packagesLogger.info(`✨ Transform the free text input into: ${JSON.stringify(params)}`);
        }
```

Mostly changed the type of request body to be of `PackagesGenerationParamsWithStringDatesAndFreeText` and then no need of making `: any` for `rest` and then you can also remove return type from the `transformFreeTextIntoPackagesGenerationParams` function